crypto/s2n_ecdsa.c

```diff
@@ -158,3 +159,15 @@ int s2n_ecdsa_pkey_init(struct s2n_pkey *pkey) {
     pkey->check_key = &s2n_ecdsa_check_key_exists;
     return 0;
 }
+
+int s2n_ecdsa_pkey_matches_curve(const struct s2n_ecdsa_key *ecdsa_key, const struct s2n_ecc_named_curve *curve)
+{
+    notnull_check(ecdsa_key);
+    notnull_check(ecdsa_key->ec_key);
+    notnull_check(curve);
+
+    int curve_id = EC_GROUP_get_curve_name(EC_KEY_get0_group(ecdsa_key->ec_key));
+    eq_check(curve_id, curve->libcrypto_nid);
```

Comment:
Is there a better error that could be returned here? The unit tests EXPECT_FAILURE(), and I didn't see how the integ tests would know this is a problem. Is there an easy way for a customer to see they are using the wrong curve in the certificate?

Reply:
I could return a better error here, but it wouldn't be surfaced. This function is used to ensure that we DON'T choose the wrong sig scheme, not to throw an error if we do. If there are no sig schemes that are valid (including having the right curve), then we throw the standard "S2N_ERR_INVALID_SIGNATURE_SCHEME".